File: Source/WebDemo/SerenePSM/SerenePSM.Web/Modules/Default/Pfk7411/Pfk7411Form.cs
using Serenity.ComponentModel;

namespace SerenePSM.Default.Forms;

[FormScript("Default.Pfk7411")]
[BasedOnRow(typeof(Pfk7411Row), CheckNames = true)]
public class Pfk7411Form
{
    public string DevelopmenetCode { get; set; }
    public string IdCard { get; set; }
    public string IdhrCode { get; set; }
    public string UserId { get; set; }
    public string PassWord { get; set; }
    public string Name { get; set; }
    public string ForeignName { get; set; }
    public string SeSite { get; set; }
    public string CdSite { get; set; }
    public string SeMainProcess { get; set; }
    public string CdMainProcess { get; set; }
    public string SeSubProcess { get; set; }
    public string CdSubProcess { get; set; }
    public string SeNationality { get; set; }
    public string CdNationality { get; set; }
    public string SeDepartment { get; set; }
    public string CdDepartment { get; set; }
    public string SeOnePosition { get; set; }
    public string CdOnePosition { get; set; }
    public string SePosition { get; set; }
    public string CdPosition { get; set; }
    public string SeInCharge { get; set; }
    public string CdInCharge { get; set; }
    public string SeFactory { get; set; }
    public string CdFactory { get; set; }
    public string SeLineProd { get; set; }
    public string CdLineProd { get; set; }
    public string Email { get; set; }
    public string Mobile { get; set; }
    public string Gender { get; set; }
    public string SendEmail { get; set; }
    public string SendSms { get; set; }
    public string SendFax { get; set; }
    public string Locked { get; set; }
    public string OpenCdt { get; set; }
    public string OpenPrvDays { get; set; }
    public string OpenRate { get; set; }
    public string OpenSchedule { get; set; }
    public string OpenAlert { get; set; }
    public string OpenMessage { get; set; }
    public decimal ScreenLock { get; set; }
    public string Idcd { get; set; }
    public string Dgcd { get; set; }
    public string Gnid { get; set; }
    public string DateInsert { get; set; }
    public string DateUpdate { get; set; }
    public string InchargeInsert { get; set; }
    public string InchargeUpdate { get; set; }
    public string DateActive { get; set; }
    public string DateDeactive { get; set; }
    public string CheckActive { get; set; }
    public string FirstChangePass { get; set; }
    public string LastPass { get; set; }
    public string LastPass1 { get; set; }
    public string LastPass2 { get; set; }
    public decimal FailLoginCnt { get; set; }
    public string CheckNever { get; set; }
    public string LastLogin { get; set; }
    public string LastTime { get; set; }
    public string CheckOption1 { get; set; }
    public string CheckOption2 { get; set; }
    public string CheckOption3 { get; set; }
    public string CheckOption4 { get; set; }
    public string CheckOption5 { get; set; }
    public string CheckUse { get; set; }
    public string DepName { get; set; }
    public string EmpTypeName { get; set; }
    public string PosName { get; set; }
    public string LineName { get; set; }
    public string TeamName { get; set; }
    public string PartName { get; set; }
    public string CheckDev { get; set; }
}

File: Source/WebDemo/SerenePSM/SerenePSM.Web/Modules/ServerTypes/Default/Pfk7411Form.ts
import { PrefixedContext, initFormType } from "@serenity-is/corelib";
import { StringEditor, DecimalEditor } from "serenity.corelib";

export interface Pfk7411Form {
    DevelopmenetCode: StringEditor;
    IdCard: StringEditor;
    IdhrCode: StringEditor;
    UserId: StringEditor;
    PassWord: StringEditor;
    Name: StringEditor;
    ForeignName: StringEditor;
    SeSite: StringEditor;
    CdSite: StringEditor;
    SeMainProcess: StringEditor;
    CdMainProcess: StringEditor;
    SeSubProcess: StringEditor;
    CdSubProcess: StringEditor;
    SeNationality: StringEditor;
    CdNationality: StringEditor;
    SeDepartment: StringEditor;
    CdDepartment: StringEditor;
    SeOnePosition: StringEditor;
    CdOnePosition: StringEditor;
    SePosition: StringEditor;
    CdPosition: StringEditor;
    SeInCharge: StringEditor;
    CdInCharge: StringEditor;
    SeFactory: StringEditor;
    CdFactory: StringEditor;
    SeLineProd: StringEditor;
    CdLineProd: StringEditor;
    Email: StringEditor;
    Mobile: StringEditor;
    Gender: StringEditor;
    SendEmail: StringEditor;
    SendSms: StringEditor;
    SendFax: StringEditor;
    Locked: StringEditor;
    OpenCdt: StringEditor;
    OpenPrvDays: StringEditor;
    OpenRate: StringEditor;
    OpenSchedule: StringEditor;
    OpenAlert: StringEditor;
    OpenMessage: StringEditor;
    ScreenLock: DecimalEditor;
    Idcd: StringEditor;
    Dgcd: StringEditor;
    Gnid: StringEditor;
    DateInsert: StringEditor;
    DateUpdate: StringEditor;
    InchargeInsert: StringEditor;
    InchargeUpdate: StringEditor;
    DateActive: StringEditor;
    DateDeactive: StringEditor;
    CheckActive: StringEditor;
    FirstChangePass: StringEditor;
    LastPass: StringEditor;
    LastPass1: StringEditor;
    LastPass2: StringEditor;
    FailLoginCnt: DecimalEditor;
    CheckNever: StringEditor;
    LastLogin: StringEditor;
    LastTime: StringEditor;
    CheckOption1: StringEditor;
    CheckOption2: StringEditor;
    CheckOption3: StringEditor;
    CheckOption4: StringEditor;
    CheckOption5: StringEditor;
    CheckUse: StringEditor;
    DepName: StringEditor;
    EmpTypeName: StringEditor;
    PosName: StringEditor;
    LineName: StringEditor;
    TeamName: StringEditor;
    PartName: StringEditor;
    CheckDev: StringEditor;
}

export class Pfk7411Form extends PrefixedContext {
    static readonly formKey = 'Default.Pfk7411';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!Pfk7411Form.init)  {
            Pfk7411Form.init = true;

            var w0 = StringEditor;
            var w1 = DecimalEditor;

            initFormType(Pfk7411Form, [
                'DevelopmenetCode', w0,
                'IdCard', w0,
                'IdhrCode', w0,
                'UserId', w0,
                'PassWord', w0,
                'Name', w0,
                'ForeignName', w0,
                'SeSite', w0,
                'CdSite', w0,
                'SeMainProcess', w0,
                'CdMainProcess', w0,
                'SeSubProcess', w0,
                'CdSubProcess', w0,
                'SeNationality', w0,
                'CdNationality', w0,
                'SeDepartment', w0,
                'CdDepartment', w0,
                'SeOnePosition', w0,
                'CdOnePosition', w0,
                'SePosition', w0,
                'CdPosition', w0,
                'SeInCharge', w0,
                'CdInCharge', w0,
                'SeFactory', w0,
                'CdFactory', w0,
                'SeLineProd', w0,
                'CdLineProd', w0,
                'Email', w0,
                'Mobile', w0,
                'Gender', w0,
                'SendEmail', w0,
                'SendSms', w0,
                'SendFax', w0,
                'Locked', w0,
                'OpenCdt', w0,
                'OpenPrvDays', w0,
                'OpenRate', w0,
                'OpenSchedule', w0,
                'OpenAlert', w0,
                'OpenMessage', w0,
                'ScreenLock', w1,
                'Idcd', w0,
                'Dgcd', w0,
                'Gnid', w0,
                'DateInsert', w0,
                'DateUpdate', w0,
                'InchargeInsert', w0,
                'InchargeUpdate', w0,
                'DateActive', w0,
                'DateDeactive', w0,
                'CheckActive', w0,
                'FirstChangePass', w0,
                'LastPass', w0,
                'LastPass1', w0,
                'LastPass2', w0,
                'FailLoginCnt', w1,
                'CheckNever', w0,
                'LastLogin', w0,
                'LastTime', w0,
                'CheckOption1', w0,
                'CheckOption2', w0,
                'CheckOption3', w0,
                'CheckOption4', w0,
                'CheckOption5', w0,
                'CheckUse', w0,
                'DepName', w0,
                'EmpTypeName', w0,
                'PosName', w0,
                'LineName', w0,
                'TeamName', w0,
                'PartName', w0,
                'CheckDev', w0
            ]);
        }
    }
}